plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -1291,7 +1292,7 @@ private String matchPifFileInDirectory(final String bridgeName) {
         return "";
     }
 
-    String [] _ifNamePatterns = {
+    static String [] ifNamePatterns = {
```

Comment:
@bwsw, what if we invert the logic?

Let me explain myself. We already have 13! (that is too much, but it is indeed a nice magic number) patterns to tell which ones are the valid physical interfaces names that we should look for. However, we can do the opposite. So, we can say that the physical interface is anything that is not “^vnet*”. Then, if I decide to use a physical interface named “XYZ”, I will not need to change anything else in the code.

What do you think?

Reply:
@rafaelweingartner I think it's a good extension, but suppose it's a conceptual change which requires voting from the project team. Don't you think so?
Personally, I like your approach, but people "potentially" may meet incompatibilities in their deployments if they have weird bridging configurations. 

@rhtyd @PaulAngus @DaanHoogland @wido @GabrielBrascher can give your opinions on such a change, that @rafaelweingartner suggests?

Reply:
Is it possible for people to configure/customize the "vnet" name of the interfaces? I thought that was something hard coded in KVM. That is why I thought doing the opposite would be easier.

Reply:
@rafaelweingartner they can, but not sure that they can with CloudStack. Actually, your idea that only "vnet" devices must be excluded is good but just would like to get the opinion of other persons before changing that. In theory, it can break someone's deployment even if our tests work well.

Reply:
@rafaelweingartner what I personally like is that people will no longer be limited by using "bond" or "team" names for physical interfaces, but can use other semantic names, that they like, the same is for single NICs like "public", "private" instead of eth0, eth1.

Reply:
@bwsw I think that the proposal from @rafaelweingartner  of inverting the logic is valid and gives us flexibility. I do not see any potential issue that such change would bring to us.
+1 on following that path.

Do you guys share the same thoughts @rhtyd @PaulAngus @DaanHoogland @wido @nvazquez?

Reply:
I agree with @rafaelweingartner , @GabrielBrascher , but it would be a separate issue. I am not sure of the consequences so I would leave it out of scope for this PR.